front/src/modules/rollingStock/components/RollingStockEditor/__tests__/RollingStockEditorForm.spec.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  getDefaultRollingStockMode,
+  getRollingStockEditorDefaultValues,
+  makeTractionMode,
+} from 'modules/rollingStock/helpers/utils';
+import {
+  DEFAULT_ELEC_POWER_STARTUP_TIME,
+  DEFAULT_RAISE_PANTOGRAPH_TIME,
+} from 'modules/rollingStock/consts';
+import { modifyRollingStockElectricalValues } from '../RollingStockEditorForm';
+
+const newEmptyRollingStock = getRollingStockEditorDefaultValues(null);
+
+describe('getElectricalDefaultValues', () => {
+  it('should add electricalPowerStartupTime and raisePantographTime when an electrical mode is added', () => {
+    const currentRollingStockValues = newEmptyRollingStock;
+    const prevRsEffortCurve = null;
+    const rsEffortCurve = getDefaultRollingStockMode('1000');
+    const result = modifyRollingStockElectricalValues(
+      currentRollingStockValues,
+      prevRsEffortCurve,
+      rsEffortCurve
+    );
+    const expected = {
+      ...currentRollingStockValues,
+      electricalPowerStartupTime: DEFAULT_ELEC_POWER_STARTUP_TIME,
+      raisePantographTime: DEFAULT_RAISE_PANTOGRAPH_TIME,
+    };
+    expect(result).toEqual(expected);
+  });
+  it('should NOT add anything if a thermal mode is added', () => {
+    const currentRollingStockValues = newEmptyRollingStock;
+    const prevRsEffortCurve = null;
+    const rsEffortCurve = getDefaultRollingStockMode('thermal');
+    const result = modifyRollingStockElectricalValues(
+      currentRollingStockValues,
+      prevRsEffortCurve,
+      rsEffortCurve
+    );
+    const expected = currentRollingStockValues;
+    expect(result).toEqual(expected);
+  });
+  it('should NOT modify rollingStockValues if previous effort is already electrical', () => {
+    const currentRollingStockValues = {
+      ...newEmptyRollingStock,
+      electricalPowerStartupTime: 123,
+      raisePantographTime: 456,
+    };
+    const prevRsEffortCurve = getDefaultRollingStockMode('1000');
```

Comment:
```suggestion
    const prevRsEffortCurve = electricRsEffortCurve;
```

Reply:
careful when using the same reference and modifying afterward :)